src/RA_httpthread.cpp

```diff
@@ -255,31 +257,39 @@ void RAWeb::SetUserAgentString()
 #ifndef NTSTATUS
     #define NTSTATUS long
 #endif
-    using fnRtlGetVersion = NTSTATUS(NTAPI*)(PRTL_OSVERSIONINFOEXW lpVersionInformation);
-    auto RtlGetVersion = (fnRtlGetVersion)GetProcAddress(GetModuleHandleW(L"ntdll.dll"), "RtlGetVersion");
-    if (RtlGetVersion)
     {
-        RTL_OSVERSIONINFOEXW osVersion = {};
-        RtlGetVersion(&osVersion);
-        if (osVersion.dwMajorVersion > 0)
+        using fnRtlGetVersion = NTSTATUS(NTAPI*)(PRTL_OSVERSIONINFOEXW);
+        const auto& ntModule{ ::GetModuleHandleW(L"ntdll.dll") };
+        if (ntModule)
         {
-            sUserAgent.append("WindowsNT ");
-            sUserAgent.append(std::to_string(osVersion.dwMajorVersion));
-            sUserAgent.append(".");
-            sUserAgent.append(std::to_string(osVersion.dwMinorVersion));
+            auto RtlGetVersion{
+                reinterpret_cast<fnRtlGetVersion>(::GetProcAddress(ntModule, "RtlGetVersion"))
+            };
+            RTL_OSVERSIONINFOEXW osVersion{};
+            RtlGetVersion(&osVersion);
+            if (osVersion.dwMajorVersion > 0)
+            {
+                std::ostringstream oss;
+                oss << "WindowsNT " << osVersion.dwMajorVersion << '.' << osVersion.dwMinorVersion
+                    << '.' << osVersion.dwBuildNumber;
```

Comment:
Put the build number for good measure, even XP has build numbers. Plus there's numbers so `std::ostringstream` was used.